ArduCopter/motor_test.cpp

```diff
@@ -155,15 +155,9 @@ MAV_RESULT Copter::mavlink_motor_test_start(const GCS_MAVLINK &gcs_chan, uint8_t
         } else {
             // start test
             gcs().send_text(MAV_SEVERITY_INFO, "starting motor test");
-            ap.motor_test = true;
```

Comment:
@andyp1per thanks for this.  Is it really necessary to move "ap.motor_test = true" to be lower in the function?  It's hard for me to imagine why this is helpful.  The purpose of this flag is to stop the function from being called simultaneously by the user pushing the button the GCS rapidly.  moving this line lower should increase the chance that that could happen.

Reply:
@rmackay9 the move is 100% ncessary - its current position is what causes the threading issue because it is checked concurrently

Reply:
removing the while(using_rate_thread) { delay(1); } loop, and removing the flag check in Copter::rate_controller_thread() is sufficient?

Reply:
@tridge no that does not work